server/src/main/java/org/jboss/as/server/parsing/StandaloneXml.java

```diff
@@ -46,14 +48,26 @@
  */
 public class StandaloneXml extends CommonXml {
 
-    private final ExtensionXml extensionXml;
-    private final ExtensionRegistry extensionRegistry;
+    private final StandaloneXml_4.ParsingOptions parsingOptions;
+    private ModuleLoader loader;
+    private ExecutorService executorService;
+    private ExtensionRegistry extensionRegistry;
```

Comment:
StandaloneXml_Legacy's constructor should take ExtensionHandler and then these fields are not needed.